src/cls/fifo/cls_fifo_types.h

```diff
@@ -361,61 +406,48 @@ struct info {
     return fmt::format("{}.{}", oid_prefix, part_num);
   }
 
-  journal_entry next_journal_entry(std::string tag) const {
-    journal_entry entry;
-    entry.op = journal_entry::Op::create;
-    entry.part_num = max_push_part_num + 1;
-    entry.part_tag = std::move(tag);
-    return entry;
-  }
-
-  std::optional<std::string>
-  apply_update(const update& update) {
+  bool apply_update(const update& update) {
+    bool changed = false;
+    namespace sr = std::ranges;
     if (update.tail_part_num()) {
       tail_part_num = *update.tail_part_num();
+      changed = true;
```

Comment:
how about this check 
if (update.tail_part_num() &&  *update.tail_part_num() != tail_part_num){
     tail_part_num = *update.tail_part_num();
      changed = true;
}

Reply:
Reasonable, all right.

Reply:
also do you plan to use boost::container::flat_set instead of vector ?
then all the changes can be tested together